src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -195,12 +220,31 @@ async def handle_bsp_scalac_options_request(
     )
     coarsened_targets = await Get(CoarsenedTargets, Addresses(tgt.address for tgt in targets))
     resolve = await Get(CoursierResolveKey, CoarsenedTargets, coarsened_targets)
+    lockfile = await Get(CoursierResolvedLockfile, CoursierResolveKey, resolve)
+
+    resolve_digest = await Get(
+        Digest,
+        CreateDigest([FileEntry(entry.file_name, entry.file_digest) for entry in lockfile.entries]),
+    )
+
+    resolve_digest = await Get(
+        Digest, AddPrefix(resolve_digest, f"jvm/resolves/{resolve.name}/lib")
+    )
```

Comment:
Minor nit: you could probably create the FileEntry names with the prefix directly, rather than adding it afterward.